src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -1149,6 +1162,124 @@ public bool PutCounters(DocumentsOperationContext context, string documentId, st
             }
         }
 
+        public int FixCountersForDocument(DocumentsOperationContext context, string documentId)
+        {
+            List<string> allNames = null;
+            LazyStringValue collection = default;
+
+            Table writeTable = default;
+            int numOfCounterGroupFixed = 0;
+            CollectionName collectionName = default;
+            try
+            {
+                var table = new Table(CountersSchema, context.Transaction.InnerTransaction);
+
+                using (DocumentIdWorker.GetSliceFromId(context, documentId, out Slice key, separator: SpecialChars.RecordSeparator))
+                {
+                    foreach (var result in table.SeekByPrimaryKeyPrefix(key, Slices.Empty, 0))
```

Comment:
it seems you modify this table while iterating... although you have here `writeTable`.. not sure if it is safe or not.